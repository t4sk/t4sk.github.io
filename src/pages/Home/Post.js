import React from "react"
import PropTypes from "prop-types"
import { Image } from "semantic-ui-react"
import styles from "./Post.module.css"

export function Post(props) {
  const { youTubeUrl, thumbnailUrl, title, createdAt } = props

  return (
    <a className={styles.component} href={youTubeUrl} target="__blank">
      <div className={styles.image}>
        <Image src={thumbnailUrl} fluid />
      </div>
      <div className={styles.content}>
        <div className={styles.title}>{title}</div>
        <div className={styles.date}>{createdAt}</div>
      </div>
    </a>
  )
}

Post.propTypes = {
  youTubeUrl: PropTypes.string.isRequired,
  thumbnailUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
}

export default Post
