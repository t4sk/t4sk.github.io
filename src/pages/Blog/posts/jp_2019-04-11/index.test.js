import React from "react"
import { shallow } from "enzyme"

import Post from "./index"

test("it renders", () => {
  const component = shallow(<Post />)

  expect(component).toMatchSnapshot()
})
