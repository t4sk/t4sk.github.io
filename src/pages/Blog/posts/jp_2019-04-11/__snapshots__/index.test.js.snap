// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<PostLayout
  date="2019-04-11"
  lang="jp"
  title="Solidity 入門 - パート 0 - MetaMaskの設定"
>
  <p>
    これは 
    <a
      href="https://solidity-by-example.org"
      rel="noopener noreferrer"
      target="__blank"
    >
      Solidity By Example
    </a>
     からのシリーズの最初の投稿です。
    <a
      href="https://solidity.readthedocs.io"
      rel="noopener noreferrer"
      target="__blank"
    >
      Solidity
    </a>
     を使ってスマートコントラクトのプログラミングを一緒に学びましょう。
  </p>
  <p>
    Ethereum のブロックチェーンでスマートコントラクトを動かすのに Ether が必要です。 イーサリアムのウォレット 
    <a
      href="https://metamask.io"
      rel="noopener noreferrer"
      target="__blank"
    >
      MetaMask
    </a>
     をインストールしましょう。
  </p>
  <h2
    id="metamask-メタマスク-の設定"
  >
    MetaMask (メタマスク) の設定
  </h2>
  <hr />
  <p>
    まずはじめに 
    <a
      href="https://metamask.io"
      rel="noopener noreferrer"
      target="__blank"
    >
      MetaMask
    </a>
     をインストールしてください。
  </p>
  <p>
    <Image
      alt="welcome to metamask"
      src="1_welcome.png"
    />
  </p>
  <p>
    ブラウザ右上にメタマスクのアイコンが表示されます。
  </p>
  <p>
    <Image
      alt="metamask icon"
      src="2_meta_mask_icon.png"
    />
  </p>
  <p>
    <code>
      CREATE A WALLET
    </code>
    をクリックしてパスワードを入力してください。
  </p>
  <p>
    <Image
      alt="create a wallet"
      src="3_create_wallet.png"
    />
  </p>
  <p>
    <strong>
      重要
    </strong>
     12 の英単語（シード）をバックアップしてください。
  </p>
  <p>
    <Image
      alt="create a wallet"
      src="5_seed.png"
    />
  </p>
  <p>
    おめでとうございます。これで MetaMask のセットアップは完了です。
  </p>
  <p>
    <Image
      alt="account"
      src="7_account.png"
    />
  </p>
  <h2
    id="テストネットワークから-ether-の入手"
  >
    テストネットワークから Ether の入手
  </h2>
  <hr />
  <p>
    それでは、テストネットワークで無料の Ether を入手しましょう。
    <em>
      Ropsten Test Network
    </em>
     を選択してください。
  </p>
  <p>
    <Image
      alt="ropsten test net"
      src="8_ropsten.png"
    />
  </p>
  <p>
    <code>
      DEPOSIT
    </code>
    をクリックしてください。
  </p>
  <p>
    <Image
      alt="deposit"
      src="9_deposit.png"
    />
  </p>
  <p>
    次に 
    <code>
      GET ETHER
    </code>
     をクリックしてください。
  </p>
  <p>
    <Image
      alt="get ether"
      src="10_get_ether.png"
    />
  </p>
  <p>
    <code>
      request 1 ether from faucet
    </code>
    をクリックして、数分待ちます。 トランザクションへのリンクが表示され、ウォレットに Ether が追加されます。
  </p>
  <p>
    <Image
      alt="faucet"
      src="12_faucet.png"
    />
  </p>
  <p>
    <Image
      alt="balance"
      src="13_balance.png"
    />
  </p>
  <h2
    id="まとめ"
  >
    まとめ
  </h2>
  <hr />
  <p>
    <a
      href="https://metamask.io"
      rel="noopener noreferrer"
      target="__blank"
    >
      MetaMask
    </a>
     は Ethereum のウォレットで、ブロックチェーン上のスマートコントラクトとやり取りすることができます。 
    <br />
    <a
      href="https://metamask.io"
      rel="noopener noreferrer"
      target="__blank"
    >
      MetaMask
    </a>
     のインストール、
    <em>
      Ropsten
    </em>
     テストネットワークからの Ether の入手方法を紹介いたしました。
  </p>
</PostLayout>
`;
